conda/plugins/installers/pip.py

```diff
@@ -0,0 +1,19 @@
+# Copyright (C) 2012 Anaconda, Inc
+# SPDX-License-Identifier: BSD-3-Clause
+"""Register the pip installer for conda env files."""
+
+from .. import CondaInstaller, hookimpl
+
+
+@hookimpl
+def conda_installers():
+    from ...env.installers.pip import dry_run as pip_dry_run
+    from ...env.installers.pip import install as pip_install
+
+    # In this demo, swap the meaning of "pip" and "pypi"
+    yield CondaInstaller(
+        name="pypi_pip",
+        types=("pip","pypi"),
```

Comment:
These `types` are mapping to the `Environment.external_packages` dict. The environment.yaml environment spec plugin will put all pip depenencies in the `pip` key of that dict. This relationship is not well definied. It would be better if it could be more strict.